libs/token/src/src_7.sw

```diff
@@ -0,0 +1,171 @@
+library;
+
+use src_7::Metadata;
+use std::string::String;
+
+impl Metadata {
+    /// Returns the underlying metadata as a `String`.
+    ///
+    /// # Returns
+    ///
+    /// * [Option<String>] - `Some` if the underlying type is a `String`, otherwise `None`.
+    ///
+    /// # Examples
+    ///
+    /// ```sway
+    /// use std::string::String;
+    /// use token::src_7::*;
+    /// use src_7::{SRC7, Metadata};
+    ///
+    /// fn foo(contract_id: ContractId, asset: AssetId, key: String) {
+    ///     let metadata_abi = abi(SRC7, contract_id);
+    ///     let metadata = metadata_abi.metadata(asset, key);
+    ///
+    ///     let string = metadata.unwrap().as_string();
+    ///     assert(string.len() != 0);
+    /// }
+    /// ```
+    fn as_string(self) -> Option<String> {
+        match self {
+            StringData(data) => Option::Some(data),
+            _ => Option::None(),
```

Comment:
```suggestion
            _ => Option::None,
```

Aaaaa, guess who's back back back back again-gain-gain shady is back back back tell a friend friend friend, now everyone report to the dance floor, to the dance floor, to the dance floor